### PR TITLE
Adding instructions on spinning up a Node.js server

### PR DESCRIPTION
**Pull request by pacostas**
Adding info on how to setup and start a Node.js server.

### CODE REVIEW

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 1]
```diff
@@ -42,13 +42,30 @@ ifeval::[{project_product}==true]
...
+----
+    node_modules/
+----
+
+Lets make our first commit:
+----
+git add .
+git commit -m 'Initial commit'
+----
+
```
Comment (abstractj) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]: I'd suggest removing the entire block above. Cover `git` steps is out of the scope for the Keycloak, plus Node.js developers are already familiar with `.gitignore`.

Comment (pacostas) [reply securing_apps/topics/oidc/nodejs-adapter.adoc]: Yeap!

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 2]
```diff
@@ -97,6 +128,14 @@ session store that `express-session` is using.
     const session = require('express-session');
     const memoryStore = new session.MemoryStore();
 
+    // Configure session
```
Comment (abstractj) [inline securing_apps/topics/oidc/nodejs-adapter.adoc:131]: @pacostas I believe it would make sense to indent the code below. Other than that, changes lgtm

**Review by abstractj — DISMISSED** [review]
LGTM

**Comment by abstractj** [comment]
@jonkoops waiting for your review, before merging it.

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 3]
```diff
@@ -58,10 +60,39 @@ involves no arguments.
...
+----
+
 By default, this will locate a file named `keycloak.json` alongside
-the main executable of your application to initialize keycloak-specific
-settings (public key, realm name, various URLs).  The `keycloak.json` file
-is obtained from the {project_name} Admin Console.
+the main executable of your application, in our case on the root folder, to initialize keycloak-specific
+settings such as public key, realm name, various URLs.
+
+In that case a Keycloak deployment is necessary to access keycloak admin console.
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
In that case a Keycloak deployment is necessary to access Keycloak admin console.
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 4]
```diff
@@ -58,10 +60,39 @@ involves no arguments.
...
+
+Please visit links on how to deploy a keycloak admin console with
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
Please visit links on how to deploy a Keycloak admin console with 
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 5]
```diff
@@ -109,13 +150,36 @@ By default, the scope value `openid` is passed as a query parameter to {project_
...
     app.use( keycloak.middleware() );
 ----
 
+Last but not least, lets open a port and set our server, to listen for http requests on port 3000, by amending below code on server.js file.
+
+[source,javascript]
+----
+    app.listen(3000, () => {
+        console.log(`App listening on port 3000`)
+    })
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
    app.listen(3000, function () {
        console.log('App listening on port 3000');
    });
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 6]
```diff
@@ -109,13 +150,36 @@ By default, the scope value `openid` is passed as a query parameter to {project_
...
+
 
+and set keycloak middleware as middleware in express, by pasting at the bottom of the file the below code
+
+[source,javascript]
+----
     app.use( keycloak.middleware() );
 ----
 
+Last but not least, lets open a port and set our server, to listen for http requests on port 3000, by amending below code on server.js file.
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
Last but not least, let's set up our server to listen for HTTP requests on port 3000 by adding the following code to `main.js`:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 7]
```diff
@@ -109,13 +150,36 @@ By default, the scope value `openid` is passed as a query parameter to {project_
...
+then require express in our project by pasting on top of the server.js file the below code
+
 [source,javascript]
 ----
+    const express = require('express');
     const app = express();
+----
+
 
+and set keycloak middleware as middleware in express, by pasting at the bottom of the file the below code
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
and configure Keycloak middleware in Express, by adding at the code below:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 8]
```diff
@@ -49,6 +49,8 @@ The `Keycloak` class provides a central point for configuration
 and integration with your application.  The simplest creation
 involves no arguments.
 
+At the root folder of your project create a file called server.js and add below block of code.
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
In the root directory of your project create a file called `server.js` and add the following code:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 9]
```diff
@@ -58,10 +60,39 @@ involves no arguments.
     const keycloak = new Keycloak({ store: memoryStore });
 ----
 
+Add `express-session` dependency:
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
Install the `express-session` dependency:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 10]
```diff
@@ -58,10 +60,39 @@ involves no arguments.
...
+
+----
+    npm install express-session --save
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
    npm install express-session
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 11]
```diff
@@ -58,10 +60,39 @@ involves no arguments.
...
+----
+
+To start the server.js script, add on package.json in scripts section, the start script:
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
To start the `server.js` script, add the following command in the 'scripts' section of the `package.json`:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 12]
```diff
@@ -58,10 +60,39 @@ involves no arguments.
...
+
+----
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "start" "node server.js"
+    },
+----
+
+And now we have the ability to run our server with below command:
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
Now we have the ability to run our server with following command:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 13]
```diff
@@ -97,6 +128,16 @@ session store that `express-session` is using.
     const session = require('express-session');
     const memoryStore = new session.MemoryStore();
 
+    // Configure session
+    app.use(
+      session({
+        secret: "mySecret",
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
        secret: 'mySecret',
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 14]
```diff
@@ -109,13 +150,36 @@ By default, the scope value `openid` is passed as a query parameter to {project_
 
 Once instantiated, install the middleware into your connect-capable app:
 
+In order to do so, first we have to install express in our dependencies 
+----
+npm install express --save
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
    npm install express
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 15]
```diff
@@ -109,13 +150,36 @@ By default, the scope value `openid` is passed as a query parameter to {project_
 
 Once instantiated, install the middleware into your connect-capable app:
 
+In order to do so, first we have to install express in our dependencies
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
In order to do so, first we have to install Express:
```

File: securing_apps/topics/oidc/nodejs-adapter.adoc [hunk 16]
```diff
@@ -109,13 +150,36 @@ By default, the scope value `openid` is passed as a query parameter to {project_
 
 Once instantiated, install the middleware into your connect-capable app:
 
+In order to do so, first we have to install express in our dependencies 
+----
+npm install express --save
+----
+
+then require express in our project by pasting on top of the server.js file the below code
```
Comment (jonkoops) [inline securing_apps/topics/oidc/nodejs-adapter.adoc]:
```suggestion
then require Express in our project as outlined below:
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Comment by pacostas** [comment]
@jonkoops Thank you for the Review! :)

**Comment by abstractj** [comment]
@pacostas could you please squash your commits into a single one?

**Comment by pacostas** [comment]
@abstractj Squashed

**Review by abstractj — APPROVED** [review]

**Review by jonkoops — APPROVED** [review]

**Pull request merged**